### PR TITLE
Add packages manifest

### PR DESCRIPTION
**Pull request by uberhacker**
This PR addresses #332 and adds the packages manifest file packages.json.  The `crew info` command has been added to pull data from the manifest.  Also, install.sh was updated to copy packages.json to `$CREW_CONFIG_PATH`.  To test, execute `bash update.sh`.  Then type `crew info [package]` to verify the package manifest data is available.

### CODE REVIEW

**Comment by uberhacker** [comment]
Anybody get a chance to try this out yet?  @jam7, @lyxell, @thedamian, @cstrouse, @prhero?

**Comment by lyxell** [comment]
I still think it would be a lot more reasonable to use the .rb files we already have to store information.

**Comment by uberhacker** [comment]
@lyxell: The problem with that approach is you will need all users to upgrade their existing packages in order to parse the new data.  If you can outline a plan that makes sense and gets this working going forward, I'm all ears.

**Comment by lyxell** [comment]
We could use hash-sums of the package files to keep track of changes, this would also update the packages if we change the build steps without changing the package version, which would be good.

**Comment by uberhacker** [comment]
Right, but that still doesn't address how this will work going forward.  I was hoping for an outline and a plan that works. :)

**Comment by lyxell** [comment]
I could whip up a pull request if that's what you mean? @uberhacker

**Comment by uberhacker** [comment]
@lyxell: Sure, you could pull the data from packages.json.  It was A LOT of work and I hope it was appreciated.

**Comment by thedamian** [comment]
@lyxell : The problem @uberhacker addresses here is the details about the pages. This is needed

Putting that detail in the .rb file makes it easy for people doing a package but hard for the system to search (and for a search site that I plan to build) hard to parse. 

Having ONE package.json file that can be parsed by "crew" is easy (and done) and that ONE file can searched for with a basic javascript script in our github.io site or a full site to give that search that brew has.

**Comment by thedamian** [comment]
I'm ready to merge but I like this conversation.

**Comment by uberhacker** [comment]
What do you guys think about this proposal?  Why not just go ahead and use packages.json for now and we can still update all the .rb files?  Over time, we will eventually get them all populated and at that point we can start parsing the .rb files instead.  However, keep in mind, what @thedamian is stating above is very true.  It's easy (and faster) to pull data from one file rather than parsing potentially hundreds of .rb files to get a list of all the data.  Suggestion: Why not have both and the ability to automatically generate packages.json by parsing all the .rb files ftw?

**Comment by thedamian** [comment]
@uberhacker I like the idea of automatically parsing, but we'd have to have another app to do that, and it can be then run by anything and the changes can be submitted as a PR.

someone will have to make it ٩(̾●̮̮̃̾•̃̾)۶

**Comment by cstrouse** [comment]
You may want to look at how Google's done the packaging for [webports](https://chromium.googlesource.com/webports/+/master/lib/webports/bsd_pkg.py). They use the FreeBSD pkg-ng format. The manifest uses ucl (universal configuration language) and is pretty nice.

**Comment by uberhacker** [comment]
@lyxell: Were you going to create a PR that replaces my proposal?  It's not easy, I know. :)  What do you think of @cstrouse's idea of Google webports?

**Comment by uberhacker** [comment]
Hey guys,
@lyxell @jam7 @thedamian @cstrouse 
I have the entire code in place to provide package info.  A new command has been added to crew called `crew refresh`.  This function is also called anytime packages are upgraded.  It will parse all the `.rb` files in `chromebrew/packages`, extract out the package info and add it to `/usr/local/etc/crew/packages.json`.  If we can get all the info in the packages and start pushing up PRs, eventually we will have everything up to date.  Below is an example package summary with all the info populated:
```
require 'package'

class Php5 < Package
  description 'PHP is a popular general-purpose scripting language that is especially suited to web development.'
  homepage 'http://www.php.net/'
  type 'package'
  category 'development'
  version '5.6.29'
  source_url 'http://php.net/distributions/php-5.6.29.tar.xz'  # software source tarball url
  source_sha1 'df3301d11d56fbeabd8f6336e9c8f4c47af45562'       # source tarball sha1 sum
```
Let me know if this is a feasible solution or if you can think of a better way to do it.
Thanks

File: crew [hunk 1]
```diff
@@ -209,30 +239,26 @@ end
...
+    @pkg.dependencies.each do |dep|
+      if not search(dep, true)
+        if not @dependencies.include? dep
+          @dependencies.push(dep)
+        end
       end
     end
   end
-  
-  pushDeps
```
Comment (jam7) [inline crew]: Does this pushDeps refactoring works correctly?  What does it happens if A depends_on B, B depends_on C, and C depends_on D?  How did you test your modifications?

It looks like way too much modifications at once for me.

Comment (uberhacker) [inline crew]: @jam7: Thank you for the input.  We don't need packages.json but since I already spent the effort to create it, we have the data ready to pull from there to the ruby packages.  You are correct about the recursive dependencies.  My thinking was only prompt for dependency packages if they aren't already installed.  Currently, the process doesn't take that into account.  I'll keep working on it.

Comment (uberhacker) [inline crew]: @jam7: I refactored pushDeps so it recursively calls itself to drill down and get every possible dependency.  Only dependencies that have not been installed yet will be prompted.

**Comment by uberhacker** [comment]
Replaced by #719.
